### DISCUSSION TITLE
Fail to build the project with 'SSL_CTX_up_ref'

### DISCUSSION THREAD

**Discussion by hououou** [memgraph/memgraph#366] [Q&A] [answered]
Hi, it is me again. I have an issue when I do the step "make -j$(nproc) memgraph". The Information is followed. 

##########information1 here#########
memgraph/src/communication/context.cpp:148:3: error: use of undeclared identifier 'SSL_CTX_up_ref'; did you mean 'SSL_CTX_free'?
  SSL_CTX_up_ref(ctx_);//SSL_CTX_free(ctx_);//
  ^~~~~~~~~~~~~~
  SSL_CTX_free
/usr/include/openssl/ssl.h:2088:6: note: 'SSL_CTX_free' declared here
void SSL_CTX_free(SSL_CTX *);
     ^
1 error generated.
make[3]: *** [src/communication/CMakeFiles/mg-communication.dir/build.make:174: src/communication/CMakeFiles/mg-communication.dir/context.cpp.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/Makefile2:2185: src/communication/CMakeFiles/mg-communication.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:1885: src/CMakeFiles/memgraph.dir/rule] Error 2
make: *** [Makefile:394: memgraph] Error 2
##########information1 end#########

And I followed the suggestion and changed "SSL_CTX_up_ref" to "SSL_CTX_free". I built the project successfully. I am not sure that I run correctly. Can you give me some advice. Thanks.

**Comment by jbajic** [comment, marked as answer]
Hi, sorry for not mentioning this earlier, but we are currently working on the fix for building Centos 7 in the https://github.com/memgraph/memgraph/pull/361 PR. You can check out that branch, but it's gonna get merged into master soon.

**Reply by jbajic** [reply]
It's been merged to the master now, you can check it out!

**Reply by hououou** [reply]
Thanks!

**Comment by gitbuda** [comment]
A quick note from my side 😄

Our community has enthusiastically embraced [Discord](https://discord.gg/memgraph) and [Stackoverflow](https://stackoverflow.com/questions/tagged/memgraphdb) as the favored channels for sharing with our team and asking questions. We’ve decided to disable the discussions tab here on GitHub in 2 weeks.

We look forward to seeing you there!
